src/app/core/components/storage-dialog/storage-dialog.component.ts

```diff
@@ -47,15 +54,28 @@ export class StorageDialogComponent implements OnInit, IsTabbedDialog {
   }
 
   ngOnInit(): void {
-    this.route.paramMap.subscribe((params: ParamMap) => {
-      const path = params.get('path') as StorageDialogTab;
+    this.subscriptions.push(
+      this.route.paramMap.subscribe((params: ParamMap) => {
+        const path = params.get('path') as StorageDialogTab;
 
-      if (path && this.tabs.includes(path)) {
-        this.activeTab = path;
-      } else {
-        this.activeTab = 'add';
-      }
-    });
+        if (path && this.tabs.includes(path)) {
+          this.activeTab = path;
+        } else {
+          this.activeTab = 'add';
+        }
+      }),
+
+      this.route.queryParamMap.subscribe((params) => {
+        const param = params.get('promoCode');
+        if (this.activeTab === 'promo' && param === this.TELLYOURSTORY) {
+          this.promoForm.setValue({ code: this.TELLYOURSTORY });
+        }
```

Comment:
Instead of hardcoding it to only work on the `TellYourStory` code, do we want to generalize it so it can take any code? That way in the future if we want to supply a link to another code we don't have to update the code at all.

Reply:
good point!